Translation/Extractor/FileExtractor.php

```diff
@@ -53,8 +56,13 @@ public function __construct(\Twig_Environment $twig, LoggerInterface $logger, ar
         $this->twig = $twig;
         $this->logger = $logger;
         $this->visitors = $visitors;
-        $lexer = new \PHPParser_Lexer();
-        $this->phpParser = new \PHPParser_Parser($lexer);
+        $lexer = new Lexer();
+        if(class_exists('PhpParser\ParserFactory')) {
+            $factory = new ParserFactory();
+            $this->phpParser = $factory->create(ParserFactory::PREFER_PHP7,$lexer);
+        } else {
+            $this->phpParser = new \PHPParser_Parser($lexer);
+        }
 
```

Comment:
I know that the code was in the tests which makes my arguement weaker but I did not think of the options. I changed my mind. I agree with you. Do not make a factory.